test/ncengine/graphics2/win32/DiligentEngineWin32_tests.cpp

```diff
@@ -0,0 +1,434 @@
+    float3 Color : COLOR; 
+};
+
+struct PSOutput
+{ 
+    float4 Color : SV_TARGET; 
+};
+
+void main(in  PSInput  PSIn,
+          out PSOutput PSOut)
+{
+    PSOut.Color = float4(PSIn.Color.rgb, 1.0);
+}
+)";
+
+void SetupSquare(nc::graphics::DiligentEngine* engine, Diligent::IPipelineState** ppPso)
+{
+    using namespace Diligent;
+
+    GraphicsPipelineStateCreateInfo PSOCreateInfo;
+    PSOCreateInfo.PSODesc.Name = "Simple triangle PSO";
+    PSOCreateInfo.PSODesc.PipelineType = PIPELINE_TYPE_GRAPHICS;
+    PSOCreateInfo.GraphicsPipeline.NumRenderTargets             = 1;
+    PSOCreateInfo.GraphicsPipeline.RTVFormats[0]                = engine->SwapChain()->GetDesc().ColorBufferFormat;
+    PSOCreateInfo.GraphicsPipeline.DSVFormat                    = engine->SwapChain()->GetDesc().DepthBufferFormat;
+    PSOCreateInfo.GraphicsPipeline.PrimitiveTopology            = PRIMITIVE_TOPOLOGY_TRIANGLE_LIST;
+    PSOCreateInfo.GraphicsPipeline.RasterizerDesc.CullMode      = CULL_MODE_NONE;
+    PSOCreateInfo.GraphicsPipeline.DepthStencilDesc.DepthEnable = False;
+
+    ShaderCreateInfo ShaderCI;
+    ShaderCI.SourceLanguage = SHADER_SOURCE_LANGUAGE_HLSL;
+    ShaderCI.Desc.UseCombinedTextureSamplers = true;
+
+    RefCntAutoPtr<IShader> pVS;
+    {
+        ShaderCI.Desc.ShaderType = SHADER_TYPE_VERTEX;
+        ShaderCI.EntryPoint      = "main";
+        ShaderCI.Desc.Name       = "Triangle vertex shader";
+        ShaderCI.Source          = VSSource;
+        engine->Device()->CreateShader(ShaderCI, &pVS);
+    }
+
+    RefCntAutoPtr<IShader> pPS;
+    {
+        ShaderCI.Desc.ShaderType = SHADER_TYPE_PIXEL;
+        ShaderCI.EntryPoint      = "main";
+        ShaderCI.Desc.Name       = "Triangle pixel shader";
+        ShaderCI.Source          = PSSource;
+        engine->Device()->CreateShader(ShaderCI, &pPS);
+    }
+
+    PSOCreateInfo.pVS = pVS;
+    PSOCreateInfo.pPS = pPS;
+    engine->Device()->CreateGraphicsPipelineState(PSOCreateInfo, ppPso);
+}
+
+void RenderSquare(nc::graphics::DiligentEngine* engine, Diligent::IPipelineState* pPso)
+{
+    using namespace Diligent;
+
+    const float ClearColor[] = {0.350f, 0.350f, 0.350f, 1.0f};
+    auto* pRTV = engine->SwapChain()->GetCurrentBackBufferRTV();
+    auto* pDSV = engine->SwapChain()->GetDepthBufferDSV();
+    engine->Context()->SetRenderTargets(1, &pRTV, pDSV, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+    engine->Context()->ClearRenderTarget(pRTV, ClearColor, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+    engine->Context()->ClearDepthStencil(pDSV, CLEAR_DEPTH_FLAG, 1.f, 0, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+    engine->Context()->SetPipelineState(pPso);
+
+    DrawAttribs drawAttrs;
+    drawAttrs.NumVertices = 4;
+    engine->Context()->Draw(drawAttrs);
+}
+} // anonymous namespace
+
+using namespace nc::graphics;
+
+TEST(DiligentEngineWin32_tests, CreateDiligentEngine_VulkanNotHeadless_Succeeds)
+{
+    ExpectDiligentInitSucceeds(false, "vulkan", false);
```

Comment:
Are we fine saying that tests will only pass on environments that support all apis? For example, what if the surface can't initialize D3D12 or something?